src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(initializeBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find book by id :", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(initializeBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find All books: ", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int updatedRows = statement.executeUpdate();
+            isUpdated(updatedRows, book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update book :" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            int updatedRows = statement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete book by id: " + id, e);
+        }
+    }
+
+    private Book initializeBook(ResultSet resultSet) {
+        try {
+            return new Book(
+                    resultSet.getLong("id"),
+                    resultSet.getString("title"),
+                    resultSet.getBigDecimal("price"));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get data from database", e);
+        }
+    }
+
+    private void isUpdated(int updatedRows, Book book) {
```

Comment:
Rename, validateUpdate for example. 
`is.... ` is usually used with methods return boolean type